site/docs/v1/tech/developer-guide/api-gateways/kong-gateway.adoc

```diff
@@ -0,0 +1,190 @@
+
+image::api-gateways/kong-gateway/kong-gateway-fusionauth-jwt-settings.png[FusionAuth application JWT settings,width=1200]
+
+Click the [uielement]#Save# button.
+
+Now, navigate to [breadcrumb]#Settings -> Key Master#. You will see the access key that you've just created in this list. By default, it has the name `Access token signing key generated for application <NAME_OF_APPLICATION>`. View the key and copy the entire contents of the [field]#PEM encoded# field under the [breadcrumb]#Public Key# section into a file in your current working directory. Name the file something like `public.pem`
+
+image::api-gateways/kong-gateway/kong-gateway-public-key.png[FusionAuth Public Key for Access Token,width=1200]
+
+You will use FusionAuth's API to generate a test JWT for your application. To do this, you need an API key with POST access to the `/api/login` endpoint. Go to [breadcrumb]#Settings -> API Keys# to add the key. Make sure the appropriate tenant is selected in the [field]#Tenant# field.
+
+image::api-gateways/kong-gateway/kong-gateway-api-key-setup.png[FusionAuth API Key setup, width=1200]
+
+Finally, make sure there is at least one user registered to your application so that you can test with a JWT issued for that user. You can create a new user by navigating to [breadcrumb]#Users -> Add user#. Untoggle the [field]#Send email to set up password# switch to manually enter a password in the [field]#Password# field.
+
+image::api-gateways/kong-gateway/kong-gateway-create-user.png[FusionAuth User Creation, width=1200,role=bottom-cropped]
+
+After saving this user, click [uielement]#Manage# and go to the [breadcrumb]#Registrations# tab. Click [uielement]#Add Registration# to register the user to your application.
+
+image::api-gateways/kong-gateway/kong-gateway-user-registration.png[FusionAuth User Registration, width=1200]
+
+== Add a Consumer in Kong Gateway
+
+Now that FusionAuth has all the requisite information to create a JWT, you can configure your Kong Gateway and JWT plugin to provide access based on valid tokens created by FusionAuth.
+
+Execute the following command in your terminal to add a consumer to Kong.
+
+[source,shell,title="Creating a consumer"]
+----
+curl -i -X POST http://localhost:8001/consumers \
+  --data "username=fusionauth"
+----
+
+This command creates a Kong consumer with the username `fusionauth`. This consumer will be linked to the authentication information that you supply to it, namely the [field]#Public key# and [field]#Issuer# name from FusionAuth. This consumer uses the information provided to verify any JWT used to access the service. You can enable this with the following command:
```

Comment:
Can you use JWKS to have kong get the public key based on the JWT and issuer or do you have to provide it up front as you do?

Reply:
As far as I can see in the docs (https://docs.konghq.com/hub/kong-inc/jwt/), it doesn't have support for jwks. Their Auth0 example also sets the public key as we have as well.

Reply:
Makes sense. Seems like they reserve that for the OIDC plugin: https://github.com/Kong/kong/issues/8381#issuecomment-1035441828